accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy

```diff
@@ -863,6 +867,62 @@ class WiremockGroovyDslSpec extends WiremockSpec {
             e.message.contains "Query parameter 'age' can't be of a matching type: NOT_MATCHING for the server side"
 	}
 
+	def "should not allow query parameter with a different absent variation for server/client"() {
```

Comment:
@marcingrzejszczak  What would you say if it looked like:

```groovy
def "should not allow query parameter with a different absent variation for server/client"() {
		when:
			GroovyDsl.make dsl
		then:
			def e = thrown(IllegalStateException)
			e.message.contains "Absent cannot only be used only on one side"
		where:
			dsl << [
			   {
					request {
						method 'GET'
						urlPath("users") {
							queryParameters {
								parameter 'name': $(client(absent()), server(""))
							}
						}
					}
					response {
						status 200
					}
				},
				{
					request {
						method 'GET'
						urlPath("users") {
							queryParameters {
								parameter 'name': $(client(""), server(absent()))
							}
						}
					}
					response {
						status 200
					}
				},
				{
					request {
						method 'GET'
						urlPath("users") {
							queryParameters {
								parameter 'name': $(client(absent()), server(matching("abc")))
							}
						}
					}
					response {
						status 200
					}
				}
			]
	}
```
Editor highlighting is broken, the second option is to use lazy evaluation using closure:

```groovy
def "should not allow query parameter with a different absent variation for server/client"() {
		when:
			make()
		then:
			def e = thrown(IllegalStateException)
			e.message.contains "Absent cannot only be used only on one side"
		where:
			make << [

				{ ->
					GroovyDsl.make {
						request {
							method 'GET'
							urlPath("users") {
								queryParameters {
									parameter 'name': $(client(absent()), server(""))
								}
							}
						}
						response {
							status 200
						}
					}
				},

				{ ->
					GroovyDsl.make {
						request {
							method 'GET'
							urlPath("users") {
								queryParameters {
									parameter 'name': $(client(""), server(absent()))
								}
							}
						}
						response {
							status 200
						}
					}
				},

				{ ->
					GroovyDsl.make {
						request {
							method 'GET'
							urlPath("users") {
								queryParameters {
									parameter 'name': $(client(absent()), server(matching("abc")))
								}
							}
						}
						response {
							status 200
						}
					}
				}
			]
	}
```